HealthcareSystem/PatientWebApp/Controllers/UserController.cs

```diff
@@ -0,0 +1,137 @@
+using Microsoft.AspNetCore.Mvc;
+using Model.Users;
+using PatientWebApp.DTOs;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using Microsoft.Extensions.Configuration;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PatientWebApp.Controllers
+{
+    [Authorize]
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UserController : ControllerBase
+    {
+        private readonly IConfiguration _config;
+        private readonly IPatientService _patientService;
+        private readonly IAdminService _adminService;
+        public UserController(IConfiguration config, IPatientService patientService, IAdminService adminService)
+        {
+            _config = config;
+            _patientService = patientService;
+            _adminService = adminService;
+        }
+
+        [AllowAnonymous]
+        [HttpPost("authenticate")]
+        public IActionResult Authenticate([FromBody] UserCredentialsDTO userCredentialsDTO)
+        {
+            try
+            {
+                string patientToken = TryToLoginPatient(userCredentialsDTO.Username, userCredentialsDTO.Password);
+                string adminToken = TryToLoginAdmin(userCredentialsDTO.Username, userCredentialsDTO.Password);
+
+                if (patientToken == null && adminToken == null)
+                {
+                    return Unauthorized();
+                }
+                else if (patientToken != null)
+                {
+                    return Ok(patientToken);
+                }
+                else
+                {
+                    return Ok(adminToken);
+                }
+            }
+            catch (BadRequestException exception)
+            {
+                return BadRequest(exception.Message);
+            }
+            catch (DatabaseException exception)
+            {
+                return StatusCode(500, exception.Message);
+            }
+            
+        }
+
+       [AllowAnonymous]
+       [HttpGet("logged")]
+       public IActionResult GetLoggedUser([FromBody] string token)
+       {
+           var key = Encoding.ASCII.GetBytes(token);
+           var handler = new JwtSecurityTokenHandler();
+           var validations = new TokenValidationParameters
+           {
+               ValidateIssuerSigningKey = true,
+               IssuerSigningKey = new SymmetricSecurityKey(key),
+               ValidateIssuer = false,
+               ValidateAudience = false
+           };
+           var claims = handler.ValidateToken(token, validations, out var tokenSecure);
+
+           string username = claims.Claims.First(c => c.Type == ClaimTypes.NameIdentifier).Value;
+           string role = claims.Claims.First(c => c.Type == ClaimTypes.Role).Value;
+           string jmbg = claims.Claims.First(c => c.Type == "Jmbg").Value;
+
+           return Ok(new LoggedUserDTO(username, jmbg, role));
+       }
```

Comment:
What is the purpose of this method? Are you sending your JWT to the backend to fetch data every time? Why not just put the data you need in the JWT and read it from Javascript?